### ISSUE TITLE
Pick PDF file with Android 13

### ISSUE THREAD

**Issue by MSKoWa** [miguelpruivo/flutter_file_picker#1207]
Hello there,

I would like to use the file_picker to let the user import PDF files. With Android versions lower than 13 it works without any issues. On Android 13, there is no file browser opened as soon as I want to pick a file. As it seems there are permissions missing to open the file browser or rather to load PDF files. The PDF files are generated elsewhere.

Does some have advice on this? I am using version 5.2.4 of the file_picker plugin.

My AndroidManifest looks like this:

```
  <uses-permission android:name="android.permission.READ_MEDIA_IMAGES" />
  <uses-permission android:name="android.permission.READ_MEDIA_VIDEO" />
  <uses-permission android:name="android.permission.READ_MEDIA_AUDIO" />
  <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE"
    android:maxSdkVersion="32" />
  <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"
    android:maxSdkVersion="29" />
```

Best regards and thanks in advance!

**Comment by MSKoWa** [comment]
Has anyone the issue to load PDF files in Anroid 13?

**Comment by pandadom** [comment]
I have the same issue, it always throw:
Exception has occurred.
PlatformException (PlatformException(read_external_storage_denied, User did not allow reading external storage, null, null))
on Android 13. sdk 33
Please help.

**Comment by TheEagle0** [comment]
I have the same issue
_pickFile error::: PlatformException(read_external_storage_denied, User did not allow reading external storage, null, null)

**Comment by bytepoets-jhe** [comment]
Same error here on API 33

**Comment by alishahzad331** [comment]
I am facing the same issue. Any solutions yet?

**Comment by bayuleo** [comment]
I have solution for this issue.

I got the error when my project using :
- file_picker: ^4.5.1
- open_file: ^3.2.1

And then i try to update it to :
- file_picker: ^5.2.5 (latest version)
- open_file_plus: ^3.4.0 (i replace open_file to this, because open_file is abandoned [link](https://github.com/crazecoder/open_file/issues/225#issuecomment-1342633161))

hope it can help you :)

**Comment by KMantas** [comment]
Issue is with this check:
```
if (!this.permissionManager.isPermissionGranted(Manifest.permission.READ_EXTERNAL_STORAGE)) {
    this.permissionManager.askForPermission(Manifest.permission.READ_EXTERNAL_STORAGE, REQUEST_CODE);
    return;
}
```
inside FilePickerDelegate.java
according to this https://stackoverflow.com/questions/72948052/android-13-read-external-storage-permission-still-usable 
`if you set compileSdkVersion to 33, you should only get READ_EXTERNAL_STORAGE for android 10 and below. otherwise, your request permission will always fail.`

I haven't tested it yet, but this might  help:
```
double release=Double.parseDouble(android.os.Build.VERSION.RELEASE.replaceAll("(\\d+[.]\\d+)(.*)","$1"));
if (release <= 10 || android.os.Build.VERSION.SDK_INT <=32)) {
  if (!this.permissionManager.isPermissionGranted(Manifest.permission.READ_EXTERNAL_STORAGE)) {
     this.permissionManager.askForPermission(Manifest.permission.READ_EXTERNAL_STORAGE, REQUEST_CODE);
     return;
  }
}
```

**Comment by MSKoWa** [comment]
Hi all,

I got my problem resolved.... the issue was the permission handler plugin:

Apparently, the end users with Android 13 could not use the plugin because my function to call the file picking was surronded by an if(permission.isGranted) which is retrieved by the permission handler plugin calling the storage permission request (which I think must be called on sdkVersion < 33). So if you have sdkVersion 33 the plugin automatically returns an isDenied status no matter what you do (you cannot ask for permissions with that because you cannot specify something for Android 13 in the Manifest file for Android 13, for pdf files. Thus there won't be a permission to be requested resulting in "isDenied). What I now did was adding the query of the current Android version if it is an Android device, the file browser is opened without any issues:

```
final permissionResult =  await Permission.storage.request();
final androidVersion =  (await DeviceInfoPlugin().androidInfo)
                                      .version
                                      .sdkInt;
if (permissionResult.isGranted || androidVersion == 33 {
    final result = await FilePicker.platform.pickFiles(...);
}
```

works for me.
